sklearn/preprocessing/_encoders.py

```diff
@@ -651,17 +769,29 @@ def inverse_transform(self, X):
         found_unknown = {}
 
         for i in range(n_features):
-            n_categories = len(self.categories_[i])
+            if (self.drop is None) or self.drop_idx_[i] is None:
+                cats = self.categories_[i]
+            else:
+                cats = np.delete(self.categories_[i], self.drop_idx_[i])
+            n_categories = len(cats)
+            if n_categories == 0:
+                X_tr[:, i] = self.categories_[i][self.drop_idx_[i]]
+                j += n_categories
+                continue
             sub = X[:, j:j + n_categories]
-
             # for sparse X argmax returns 2D matrix, ensure 1D array
             labels = np.asarray(_argmax(sub, axis=1)).flatten()
-            X_tr[:, i] = self.categories_[i][labels]
-
-            if self.handle_unknown == 'ignore':
-                # ignored unknown categories: we have a row of all zero's
-                unknown = np.asarray(sub.sum(axis=1) == 0).flatten()
-                if unknown.any():
+            X_tr[:, i] = cats[labels]
+            unknown = np.asarray(sub.sum(axis=1) == 0).flatten()
```

Comment:
I think you want `~sub.any(axis=1)`

Reply:
Hrm, this doesn't play nicely with CSR matrices. Do you think it's worth converting a sparse representation to a dense one to use this?